### PR TITLE
Allow to set sudo mode for api requests

### PR DESCRIPTION
**Pull request by acasajus**

### CODE REVIEW

File: app/api/views/sudo.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    """
+    Enter sudo mode
+
+    Input
+    - password: user password to validate request to enter sudo mode
+    """
+    user = g.user
+    data = request.get_json() or {}
+    if "password" not in data:
+        return make_response(jsonify(error="Invalid request"), 403)
```
Comment (nguyenkims) [inline app/api/views/sudo.py:20]: I think we don't need `make_response` here

File: app/api/views/sudo.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+from flask import jsonify, g, request, make_response
+from sqlalchemy_utils.types.arrow import arrow
+
+from app.api.base import api_bp, require_api_auth
+from app.db import Session
+
+
+@api_bp.route("/sudo", methods=["PATCH"])
```
Comment (nguyenkims) [inline app/api/views/sudo.py:8]: should we add this endpoint to api.md ?

File: app/api/views/sudo.py (hunk 1 above)
Comment (acasajus) [reply app/api/views/sudo.py]: Removed.

File: app/api/views/sudo.py (hunk 2 above)
Comment (acasajus) [reply app/api/views/sudo.py:8]: Added. We really need at some point to generate the docs automatically from the methods.

File: app/api/views/sudo.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    data = request.get_json() or {}
+    if "password" not in data:
+        return make_response(jsonify(error="Invalid request"), 403)
+    if not user.check_password(data["password"]):
+        return make_response(jsonify(error="Invalid request"), 403)
+
+    g.api_key.sudo_mode_at = arrow.now()
+    Session.commit()
+
+    return jsonify({"ok": True})
```
[the remark this answers is not in the document]
Comment (acasajus) [reply app/api/views/sudo.py]: Done

File: docs/api.md [hunk 4]
```diff
@@ -10,6 +10,7 @@
 - [POST /api/auth/reactivate](##post-apiauthreactivate): Request a new activation code.
 - [POST /api/auth/forgot_password](#post-apiauthforgot_password): Request reset password link.
 - [GET /api/user_info](#get-apiuser_info): Get user's information.
+- [PATCH /api/sudo](#get-apiuser_sudo): Enable sudo mode.
```
Comment (nguyenkims) [inline docs/api.md:13]: Nits: should `#get-apiuser_sudo` be `#patch-apiuser_sudo`?

**Review by nguyenkims — APPROVED** [review]
LGTM! Just left a nits.

File: docs/api.md (hunk 4 above)
Comment (acasajus) [reply docs/api.md:13]:
```suggestion
- [PATCH /api/sudo](#patch-apiuser_sudo): Enable sudo mode.
```

**Pull request merged**